cpp/src/arrow/acero/concurrent_queue_internal.h

```diff
@@ -35,8 +35,7 @@ class ConcurrentQueue {
   //
   T Pop() {
     std::unique_lock<std::mutex> lock(mutex_);
-    cond_.wait(lock, [&] { return !queue_.empty(); });
-    return PopUnlocked();
+    return PopUnlocked(lock);
```

Comment:
The comment for `Pop` says it must be called on a non-empty queue?

Reply:
Hmm. Didn't See that one. But still i think it is just wrong comment. Original `ConcurrentQueue::Pop` does check for emptiness despite the comment.  Original `BackpressureConcurrentQueue:Pop`does not check for emptiness and there is no comment.

Reply:
Hmm. Do you have really meet some problem here?

Reply:
I encountered a problem when I reused `BackpressureConcurrentQueue` in new acero node (will PR soon) and pop on empty obviously caused segfaut. That was when I switched from `ConcurrentQueue` to `BackpressureConcurrentQueue` - was not expecting that. I think to the very least to comment should be moved form `ConcurrentQueue::Pop` (misleading) to `BackpressureConcurrentQueue:Pop` (missing), although I think unifying safe version of `Pop` makes much more sense. More verbose method like `PopUnsafe` can be added also if any regression occurs. LMK the way forward and I will update PR.